inc/fiber_prompt.h

```diff
@@ -0,0 +1,48 @@
+/** A basic fiber interface. **/
+#ifndef WASMFX_FIBER_C_H
+#define WASMFX_FIBER_C_H
+
+#include <stdint.h>
+
+#define export(NAME) __attribute__((export_name(NAME)))
+
+/** The abstract type of a prompt object **/
+typedef uint32_t prompt_t;
+
+/** The signature of a fiber entry point. **/
+typedef void* (*fiber_entry_point_t)(void*, prompt_t);
+
+/** The abstract type of a fiber object. **/
+typedef struct fiber* fiber_t;
+
+/** Allocates a new fiber with the default stack size. **/
+export("fiber_alloc") fiber_t fiber_alloc(fiber_entry_point_t entry);
+
+/** Reclaims the memory occupied by the fiber object. **/
+export("fiber_free") void fiber_free(fiber_t fiber);
+
+/** Yields control to the context of the provided prompt. This function must be
+   called from within a fiber context. **/
+export("fiber_yield_to") void* fiber_yield_to(void* arg);
```

Comment:
This isn't right. You definitively need to take a prompt as parameter here.
```suggestion
export("fiber_yield_to") void* fiber_yield_to(prompt_t prompt, void* arg);
```